homeassistant/components/tahoma/switch.py

```diff
@@ -0,0 +1,95 @@
+"""Support for TaHoma switches."""
+import logging
+
+from homeassistant.components.switch import (
+    DEVICE_CLASS_SWITCH,
+    DOMAIN as SWITCH,
+    SwitchEntity,
+)
+from homeassistant.const import STATE_OFF, STATE_ON
+
+from .const import COMMAND_OFF, COMMAND_ON, CORE_ON_OFF_STATE, DOMAIN
+from .tahoma_entity import TahomaEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+COMMAND_CYCLE = "cycle"
+COMMAND_MEMORIZED_VOLUME = "memorizedVolume"
+COMMAND_RING_WITH_SINGLE_SIMPLE_SEQUENCE = "ringWithSingleSimpleSequence"
+COMMAND_SET_FORCE_HEATING = "setForceHeating"
+COMMAND_STANDARD = "standard"
+
+IO_FORCE_HEATING_STATE = "io:ForceHeatingState"
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the TaHoma sensors from a config entry."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    coordinator = data["coordinator"]
+
+    entities = [
+        TahomaSwitch(device.deviceurl, coordinator)
+        for device in data["platforms"][SWITCH]
+    ]
+
+    async_add_entities(entities)
+
+
+class TahomaSwitch(TahomaEntity, SwitchEntity):
+    """Representation of a TaHoma Switch."""
+
+    @property
+    def device_class(self):
+        """Return the class of the device."""
+        return DEVICE_CLASS_SWITCH
+
+    @property
+    def is_on(self):
+        """Get whether the switch is in on state."""
+        return (
+            self.executor.select_state(CORE_ON_OFF_STATE, IO_FORCE_HEATING_STATE)
+            == STATE_ON
+        )
+
+    async def async_turn_on(self, **_):
+        """Send the on command."""
+        if self.executor.has_command(COMMAND_ON):
```

Comment:
Does it make sense to have multiple classes with a base class that has the shared functionality instead of a single class that checks each time?

Reply:
We do this for climate and water_heater currently, since these devices are very complicated and different compared to each other. The reason why we didn't do this for switch / cover is because there are quite some different types with a different command set, which means that we would need to make quite some classes.

Let me have a better look at this again :).

Reply:
Circling back on this, there are a few PRs open for this but I'm not sure where things are going with this so I've been skipping them over when looking at the review queue.

Reply:
@bdraco understandable! I have been looking into these issues and will push changes soon, we had some issues with the rebasing and figuring out the best changes.. 

However, eventually best would be to see if we can get https://github.com/home-assistant/core/pull/49165 in core and than `rework-tahoma` in core. Core is moving fast and `rework-tahoma` gets behind fast, thus I would love to see the first version merged in, where the other platforms (like switch) can follow later.